src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -667,11 +667,12 @@ public SCMFile getRoot() {
 
     @Override
     public void afterSave() {
-        GitLabWebhookRegistration mode = new GitLabSCMSourceContext(null, SCMHeadObserver.none())
-                .withTraits(new GitLabSCMNavigatorContext().withTraits(traits).traits())
-                .webhookRegistration();
+        GitLabSCMSourceContext ctx = new GitLabSCMSourceContext(null, SCMHeadObserver.none())
+                .withTraits(new GitLabSCMNavigatorContext().withTraits(traits).traits());
+        GitLabWebhookRegistration mode = ctx.webhookRegistration();
+        boolean systemHookDisabled = ctx.systemHookDisabled();
         LOGGER.info("Mode of web hook: " + mode.toString());
-        GitLabWebhookCreator.register(this, mode);
+        GitLabWebhookCreator.register(this, mode, systemHookDisabled);
```

Comment:
Yup seems counter-intuitive here that register that systemHookDisabled, when true enables system hook via isAdmin, is set to true :)